src/tests/ftest/io/posix_stat.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2018-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from avocado.core.exceptions import TestFail
+
+from ior_test_base import IorTestBase
+from general_utils import get_remote_file_size, run_task
+from apricot import TestWithServers
+
+
+class POSIXStatTest(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test class description:
+
+    Requirement: SRS-10-0303
+    DFS stat call reports meaningful information (i.e. valid file size, correct
+    block size, ...).
+
+    Test Steps:
+    Create files of different size (block size) in a mounted POSIX container and
+    verify the size with stat command. Also verify the creation time with date
+    command.
+
+    :avocado: recursive
+    """
+    def test_stat_parameters(self):
+        """JIRA ID: DAOS-3769
+
+        Create files of 1M, 10M, 100M, 500M, and verify the size and creation
+        time.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=small
+        :avocado: tags=stat_parameters
+        """
+        block_sizes = self.params.get("block_sizes", "/run/*")
+        testfile_path_base = "/tmp/daos_dfuse/testfile"
+        error_list = []
+
+        self.add_pool(connect=False)
+        self.add_container(pool=self.pool)
+
+        i = 1
+        for block_size in block_sizes:
+            self.log.info("Block Size = %s", block_size)
+            self.ior_cmd.block_size.update(block_size)
+
+            # 1. Verify creation time.
+            # Get current epoch before running ior. The timestamp of the file
+            # created by ior is near the time of the start of the ior command
+            # execution.
+            current_epoch = -1
+            cmd_output = ""
+            # Run date command in the client node because that's where the file
+            # is created.
+            task = run_task(self.hostlist_clients, "date +%s")
+            for output, _ in task.iter_buffers():
+                cmd_output = "\n".join(
+                    [line.decode("utf-8") for line in output])
+            current_epoch = cmd_output.split()[-1]
+
+            test_file_suffix = "_{}".format(i)
+            i += 1
+
+            # Run ior command.
+            try:
+                self.run_ior_with_pool(
+                    timeout=200, stop_dfuse=False, create_pool=False,
+                    create_cont=False, test_file_suffix=test_file_suffix)
+            except TestFail:
+                self.log.info("ior command failed!")
+
+            # Get epoch of the created file.
+            creation_epoch = -1
+            cmd_output = ""
+            # As in date command, run stat command in the client node.
+            testfile_path = testfile_path_base + test_file_suffix
+            task = run_task(
+                self.hostlist_clients, "stat -c%Z {}".format(testfile_path))
```

Comment:
The `self.run_ior_with_pool()` method updates the `self.ior_cmd.test_file` property with the `self.dfuse.mount_dir.value` + `"testfile"` as part of it's execution, so this method does not need to track/create it:
```suggestion
            task = run_task(
                self.hostlist_clients, "stat -c%Z {}".format(self.ior_cmd.test_file.value))
```